client/apps/edit/components/content2/section_tool/index.jsx

```diff
@@ -0,0 +1,146 @@
+const IconEditText = components.Icon.EditText
+const IconEditVideo = components.Icon.EditVideo
+
+const IconHeroImage = components.Icon.HeroImage
+const IconHeroVideo = components.Icon.HeroVideo
+
+export default class SectionTool extends Component {
+  constructor (props) {
+    super(props)
+
+    this.type = {displayName: 'SectionTool'}
+    this.state = {
+      open: false
+    }
+  }
+
+  toggleOpen = () => {
+    this.setState({open: !this.state.open })
+  }
+
+  getProps(type) {
+    switch (type) {
+      case 'video':
+        return {
+          type: 'video',
+          url: '',
+          layout: 'column_width'
+        }
+      case 'image_collection':
+        return {
+          type: 'image_collection',
+          layout: 'overflow_fillwidth',
+          images: []
+        }
+      case 'embed':
+        return {
+          type: 'embed',
+          url: '',
+          layout: 'column_width',
+          height: ''
+        }
+      case 'text':
+        return {
+          type: 'text',
+          body: ''
+        }
+    }
+  }
+
+  newSection = (type) => {
+    this.props.sections.add(
+      this.getProps(type),
+      {at: this.props.index + 1}
+    )
+    this.setState({open: false})
+  }
+
+  setHero = (type) => {
+    this.props.section.set(this.getProps(type))
+    this.props.onSetEditing(true)
+  }
+
+  renderHeroMenu() {
+    if (this.state.open) {
+      return (
+        <ul className='edit-tool__menu'>
+          <li
+            className='edit-tool__hero-image'
+            onClick={ () => this.setHero('image_collection')}>
+            <IconHeroImage />
+            Large Format Image
+          </li>
+          <li
+            className='edit-tool__hero-video'
+            onClick={ () => this.setHero('video')}>
+            <IconHeroVideo />
+            Large Format Video
+          </li>
+        </ul>
+      )
+    }
+  }
+
+  renderSectionMenu() {
+    if (this.state.open) {
+      return (
+        <ul className='edit-tool__menu'>
+          <li
+            className='edit-tool__edit-text'
+            onClick={ () => this.newSection('text')}>
+            <IconEditText />
+            Text
+          </li>
+          <li
+            className='edit-tool__edit-images'
+            onClick={ () => this.newSection('image_collection')}>
+            <IconEditImages />
+            Images
+          </li>
+          <li
+            className='edit-tool__edit-video'
+            onClick={ () => this.newSection('video')}>
+            <IconEditVideo />
+            Video
+          </li>
+          <li
+            className='edit-tool__edit-embed'
+            onClick={ () => this.newSection('embed')}>
+            <IconEditEmbed />
+            Embed
+          </li>
+        </ul>
+      )
+    }
+  }
+
+  render() {
+    const { isEditing, isHero, sections } = this.props
+    const isFirstSection = this.props.firstSection && sections.length === 0 ? ' first' : ''
```

Comment:
Also, a place where a namespace would be better as `.first` is a very generic name for a css class that could accidentally conflict down the line. Alternatively, the data-attr pattern [you use below](https://github.com/artsy/positron/pull/1297/files#diff-a3b01687d52937b54c16e5ab3a27f233R129) is another way to avoid the problem because it's very unlikely a CSS selector would be written like `[data-state-open] { color: red }`.

Now that we're embarking more full-on-react-world, it might be good to peak at how we mix stylus and React in more greenfield projects like [Prediciton](https://github.com/artsy/prediction/blob/master/src/components/LotEventList.styl#L3) which puts a namespace at the top level and strictly uses the `&_modifying_class` underneath it.